Add conda-forge installation documentation for Linux and MacOSX (#1204)
nest/nest-simulator: doc/installation/conda_install.rst

If you don't have Conda installed, we recommend you follow the instructions on `Miniconda <https://conda.io/miniconda.html>`__.

If you need to install NEST with additional configuration options,
please see our :doc:`install guide for Linux <linux_install>` or :doc:`install guide for macOS <mac_install>`

.. admonition:: Important!

   The conda-forge package of `nest-simulator` is still being tested, let us know if you come across
   any problems by `submitting an issue on GitHub <https://github.com/nest/nest-simulator/issues>`_.

----

Install the latest conda-forge package for NEST
-----------------------------------------------------

1. Create your conda environment. We recommend that you create a dedicated
   environment for NEST, which should ensure there are no conflicts with previously
   installed packages.

.. code-block:: sh

  conda create --name ENVNAME

2. Activate your environment:

.. code-block:: sh

  conda activate ENVNAME

3. Install the latest `nest-simulator` package from conda-forge. You can choose to install it

WITHOUT openmpi:

.. code-block:: sh

  conda install -c conda-forge nest-simulator

or WITH openmpi:

.. code-block:: sh

>    conda install -c conda-forge nest-simulator=*=mpi_openmpi*

Reviewer: This looks like two ==, one outside and one inside *...*, unless I do not understand RST syntax right.

Author: The meaning behind the syntax is as follows:
`conda install -c conda-forge nest-simulator=<version>=<build_string>`

We could include that as an explanation if needed?

Author: ```suggestion
   conda install -c conda-forge nest-simulator=*=mpi_openmpi*
   
Where the syntax for this install follows the pattern: ``nest-simulator=<version>=<build_string>``
```

.. note::

>    The current version on conda-forge is NEST 2.16.0

Reviewer: May need updating soon?

Author: Yep I didn't want to leave this in as is but was thinking we should include some information about the versions available on conda-forge. Perhaps something along the lines of "The conda-forge package is available from NEST 2.16.0 onward"?

Author: ```suggestion
   The conda-forge package is available from NEST 2.16.0 onward
```
